# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLFormElement.h
# Commit: b9e87fde31f7
# Full Hash: b9e87fde31f7ca31185081dc96fa937927f1879f
# Author: sefeng <sefeng@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1557303
# File Overlap Count: 1
# Description:
#   Bug 1557303 - Implement form[method="dialog"] r=smaug
#   
#   Spec: https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submit-dialog
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73364
# ==============================================================================

diff -r e02018dfce9e -r b9e87fde31f7 dom/html/HTMLFormElement.h
--- a/dom/html/HTMLFormElement.h	Tue May 12 00:11:12 2020 +0000
+++ b/dom/html/HTMLFormElement.h	Mon May 11 22:56:13 2020 +0000
@@ -462,6 +462,12 @@
   nsresult SubmitSubmission(HTMLFormSubmission* aFormSubmission);
 
   /**
+   * Submit a form[method=dialog]
+   * @param aFormSubmission the submission object
+   */
+  nsresult SubmitDialog(DialogFormSubmission* aFormSubmission);
+
+  /**
    * Notify any submit observers of the submit.
    *
    * @param aActionURL the URL being submitted to